Bottom: f2e595547fd0f4870bc94d4a5ad672ac274007f0
Top:    c0e5d36100b473d1b729e9582d31d057b2886502
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-08-24 18:55:33 +0100

Attempt to debug islandness after inserting hole bug


---

diff --git a/src/action.c b/src/action.c
index 6121665..2f306f8 100644
--- a/src/action.c
+++ b/src/action.c
@@ -1507,12 +1507,19 @@ NotifyMode (void)
 		   */
 		  SaveUndoSerialNumber ();
 		  Flags = ((PolygonType *)Crosshair.AttachedObject.Ptr2)->Flags;
-		  PolyToPoursOnLayer (PCB->Data, Crosshair.AttachedObject.Ptr1,
-				      result, Flags);
+#warning FIXME: Removing a pour should force a re-check of islanding!
+                  /* For now, do this before PolyToPours() which re-inits the
+                   * clipping for the pour(s) affected (?). Ideally RemoveObject
+                   * should fix any island response its self. Anyway.. removing
+                   * the old thing first saves us doing twice the work by
+                   * recomputing islands when it gets deleted.
+                   */
 		  RemoveObject (POUR_TYPE,
 				Crosshair.AttachedObject.Ptr1,
 				Crosshair.AttachedObject.Ptr2,
 				Crosshair.AttachedObject.Ptr3);
+		  PolyToPoursOnLayer (PCB->Data, Crosshair.AttachedObject.Ptr1,
+				      result, Flags);
 		  RestoreUndoSerialNumber ();
 		  IncrementUndoSerialNumber ();
 		  Draw ();
